avrgirl-arduino.js

```diff
@@ -129,6 +129,8 @@ Avrgirl_arduino.prototype._upload = function (hex, callback) {
         self._uploadAVR109(eggs, cb);
     });
   }
+  // this is a success callback
+  callback();
```

Comment:
The callbacks will occur for this method within each protocol's upload call. A callback at the bottom of this method could potentially cause a false (and early) success. Perhaps if it doesn't find the correct protocol, a callback is returned with an error instead? Within an `else` at the end of the list of conditionals.

Let me know if I'm missing something here or the above suggestion is not the solution for what you were trying to solve.